File: Python/704BinarySearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r, = 0, len(nums) - 1
        while l <= r:
            m = (l + r) // 2
            if nums[m] > target:
                # go in left half, update right
                r = m - 1
            elif nums[m] < target:
                # go in right half, update left
                l = m + 1
            else:
                return m

        return -1

File: Swift/704BinarySearch.swift
class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        var left = 0, right = nums.count - 1
        while left <= right {
            let mid = left + (right - left) / 2
            if nums[mid] == target {
                return mid
            }
            else if target > nums[mid] {
                // go right half
                left = mid + 1
            }
            else {
                // go left
                right = mid - 1
            }
        }
        return -1
    }
}

